src/openrct2/core/Zip.cpp

```diff
@@ -31,7 +31,7 @@ class ZipArchive final : public IZipArchive
         }
 
         int32_t error;
-        _zip = zip_open(path.data(), zipOpenMode, &error);
+        _zip = zip_open(std::string(path).data(), zipOpenMode, &error);
```

Comment:
You should use c_str() to if you want null terminated string, I know you can argue that most implementations do same as data() but for correctness it should be c_str()

This should be changed in all cases.

Reply:
Actually as of C++11, c_str() and data() are exactly the same by standard ;)

Reply:
I still don't like the mixed usage, c_str() is preferred to clarify intent, when data() is used its usually assumed that its buffer is used as raw data.

Reply:
Fair point there 👍 
I will fix them

Reply:
When dealing with `string_view`, it deliberately does not have c_str as it might not be null terminated, you never know with `data()`.